packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts

```diff
@@ -220,39 +224,64 @@ export class GraphQLSitemapService {
     languages: string[],
     formatStaticPath: (path: string[], language: string) => StaticPath
   ): Promise<StaticPath[]> {
+    let paths = new Array<StaticPath>();
     if (!languages.length) {
       throw new RangeError(languageError);
     }
-    // Fetch paths using all locales
-    const paths = await Promise.all(
-      languages.map((language) => {
-        if (language === '') {
-          throw new RangeError(languageEmptyError);
-        }
-        debug.sitemap('fetching sitemap data for %s', language);
-        return this.fetchLanguageSitePaths(language).then((results) =>
-          this.transformLanguageSitePaths(results, formatStaticPath, language)
-        );
-      })
-    );
+    // Get all sites
+    const sites = this.options.sites;
+    let aggregatedPaths: StaticPath[] = [];
+
+    // Fetch paths for each site
+    for (let i = 0; i < sites?.length; i++) {
+      const siteName = sites[i];
+      const site = sites?.length > 1 ? siteName : undefined;
+      const isMultiSite = sites?.length > 1 && true;
+      const multipleLanguages = languages.length > 1 && true;
+
+      // Fetch paths using all locales
+      paths = ((await Promise.all(
+        languages.map(async (language) => {
+          if (language === '') {
+            throw new RangeError(languageEmptyError);
+          }
+          debug.sitemap('fetching sitemap data for %s', language);
+          const results = await this.fetchLanguageSitePaths(language, siteName);
+          if (multipleLanguages) {
+            aggregatedPaths = [];
+          }
+          return await this.transformLanguageSitePaths(
+            results,
+            formatStaticPath,
+            language,
+            aggregatedPaths,
+            isMultiSite,
+            site
+          );
+        })
+      )) as []) as StaticPath[];
```

Comment:
Do you need to provide specific type? `transformLanguageSitePaths` returns `StaticPath[]`, so it's should be inferred